### PR TITLE
[fix](Insert) fix 5 concurrent "insert...select..." OOM

### PR DESCRIPTION
**Pull request by yiguolei**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/sink/vtablet_sink.cpp [hunk 1]
```diff
@@ -426,6 +434,15 @@ Status VOlapTableSink::send(RuntimeState* state, vectorized::Block* input_block)
     if (findTabletMode == FindTabletMode::FIND_TABLET_EVERY_BATCH) {
         _partition_to_tablet_map.clear();
     }
+
+    //if pending bytes is more than 500M, wait
+    constexpr size_t MAX_PENDING_BYTES = 500 * 1024 * 1024;
+    if (get_pending_bytes() > MAX_PENDING_BYTES) {
+        while (get_pending_bytes() < MAX_PENDING_BYTES) {
```
Comment (yinzhijian) [inline be/src/vec/sink/vtablet_sink.cpp:441]: The if statement seems unnecessary, and the while statement's condition seems misjudged

File: be/src/vec/sink/vtablet_sink.cpp [hunk 2]
```diff
@@ -426,6 +434,15 @@ Status VOlapTableSink::send(RuntimeState* state, vectorized::Block* input_block)
...
+            std::this_thread::sleep_for(std::chrono::microseconds(500));
```
Comment (yinzhijian) [inline be/src/vec/sink/vtablet_sink.cpp:442]: Is it necessary to set a maximum waiting time?

File: be/src/vec/sink/vtablet_sink.cpp (hunk 1 above)
Comment (yiguolei) [reply be/src/vec/sink/vtablet_sink.cpp:441]: yes....

File: be/src/vec/sink/vtablet_sink.cpp (hunk 2 above)
Comment (yiguolei) [reply be/src/vec/sink/vtablet_sink.cpp:442]: I will remove these code. And add more check logic in next pr. This pr is focused cherry-pick from 1.0.1

**Comment by yinzhijian** [comment]
LGTM

**Review by caiconghui — APPROVED** [review]
LGTM

**Pull request merged**